Sources/SWBWebAssemblyPlatform/Plugin.swift

```diff
@@ -0,0 +1,144 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+public import SWBUtil
+import SWBCore
+import SWBMacro
+import Foundation
+
+@PluginExtensionSystemActor public func initializePlugin(_ manager: PluginManager) {
+    manager.register(WebAssemblyPlatformSpecsExtension(), type: SpecificationsExtensionPoint.self)
+    manager.register(WebAssemblyPlatformExtension(), type: PlatformInfoExtensionPoint.self)
+    manager.register(WebAssemblySDKRegistryExtension(), type: SDKRegistryExtensionPoint.self)
+}
+
+struct WebAssemblyPlatformSpecsExtension: SpecificationsExtension {
+    func specificationFiles() -> Bundle? {
+        .module
+    }
+
+    func specificationDomains() -> [String : [String]] {
+        ["webassembly": ["generic-unix"]]
```

Comment:
generic-unix is the base domain for non-Apple POSIX-y/UNIX-y ELF-based types of platforms. The main intent here is to share compiler and linker specs for clang and GNU/BSD linkers where the flags are generally the same across any platforms which use it (e.g. Linux and FreeBSD both inherit this).

I'm not sure if WASI fits that bill (I'm not saying it's wrong either), just want to get a sense of how similar the compiler and linker invocations are, vs whether it makes more sense to inherit from the global domain instead, and define the bits from scratch.